eng/pipelines/coreclr/templates/build-job.yml

```diff
@@ -167,7 +167,7 @@ jobs:
     # Linux builds use docker images with dependencies preinstalled,
     # and FreeBSD builds use a build agent with dependencies
     # preinstalled, so we only need this step for OSX and Windows.
-    - ${{ if eq(parameters.osGroup, 'osx') }}:
+    - ${{ if in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator') }}:
```

Comment:
No harm, although not sure this will impact any leg.

Reply:
Yes. It's easy to miss these, so I kept them in sync; since we need to run this script if it is any Apple OS && there is any native build step. We just don't have something like https://github.com/dotnet/runtime/blob/b01311bd8dc591ce26b88a86b592c63a26798ceb/eng/native/configureplatform.cmake#L102 in pipelines.

Reply:
Yes, harmless. It's easy to miss these and hard to figure out which condition is the culprit, so I kept them in sync. We need to run install-native-dependencies script in the CI, if we are on any Apple OS && there is any native build step. We just don't have something more generic like https://github.com/dotnet/runtime/blob/b01311bd8dc591ce26b88a86b592c63a26798ceb/eng/native/configureplatform.cmake#L102 in pipelines.